src/Console/Contact.php

```diff
@@ -0,0 +1,287 @@
+	bin/console contact remove <CID> [-h|--help|-?] [-v]
+	bin/console contact search id <CID> [-h|--help|-?] [-v]
+	bin/console contact search url <user nick> <URL> [-h|--help|-?] [-v]
+	bin/console contact terminate <CID> [-h|--help|-?] [-v]
+
+Description
+	Modify contact settings per console commands.
+
+Options
+    -h|--help|-? Show help information
+    -v           Show more debug information
+    -y           Non-interactive mode, assume "yes" as answer to the user deletion prompt
+HELP;
+		return $help;
+	}
+
+	public function __construct(App\Mode $appMode, array $argv = null)
+	{
+		parent::__construct($argv);
+
+		$this->appMode = $appMode;
+	}
+
+	protected function doExecute()
+	{
+		if ($this->getOption('v')) {
+			$this->out('Class: ' . __CLASS__);
+			$this->out('Arguments: ' . var_export($this->args, true));
+			$this->out('Options: ' . var_export($this->options, true));
+		}
+
+		if (count($this->args) == 0) {
+			$this->out($this->getHelp());
+			return 0;
+		}
+
+		if ($this->appMode->isInstall()) {
+			throw new RuntimeException('Database isn\'t ready or populated yet');
+		}
+
+		$command = $this->getArgument(0);
+
+		switch ($command) {
+			case 'add':
+				return $this->addContact();
+			case 'remove':
+				return $this->removeContact();
+			case 'search':
+				return $this->searchContact();
+			case 'terminate':
+				return $this->terminateContact();
+			default:
+				throw new \Asika\SimpleConsole\CommandArgsException('Wrong command.');
+		}
+	}
+
+	/**
+	 * Retrieves the user from a nick supplied as an argument or from a prompt
+	 *
+	 * @param int $arg_index Index of the nick argument in the arguments list
+	 *
+	 * @return array|boolean User record with uid field, or false if user is not found
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 */
+	private function getUserByNick($arg_index)
+	{
+		$nick = $this->getArgument($arg_index);
+
+		if (empty($nick)) {
+			$this->out('Enter user nickname: ');
+			$nick = CliPrompt::prompt();
+			if (empty($nick)) {
+				throw new RuntimeException('A user nickname must be specified.');
+			}
+		}
+
+		$user = UserModel::getByNickname($nick, ['uid', 'nickname']);
+		if (empty($user)) {
+			throw new RuntimeException('User not found');
+		}
+
+		return $user;
+	}
+
+	/**
+	 * Adds a contact to a user from a URL
+	 *
+	 * @return bool True, if the command was successful
+	 */
+	private function addContact()
+	{
+		$user = $this->getUserByNick(1);
+
+		$url = $this->getArgument(2);
+		if (empty($url)) {
+			$this->out('Enter contact URL: ');
+			$url = CliPrompt::prompt();
+			if (empty($url)) {
+				throw new RuntimeException('A contact URL must be specified.');
+			}
+		}
+
+		$contact = ContactModel::getByURLForUser($url, $user['uid']);
+		if (!empty($contact)) {
+			throw new RuntimeException('Contact already exists');
+		}
+
+		$network = $this->getArgument(3);
+		if (empty($network) && $network !== '') {
```

Comment:
Isn't `empty()` including `!== ''` as well? Seems redundant

Reply:
It's "and", not "or".  And yeah, this part is oddly written, so I'd welcome suggestions.  The idea is that an empty string is fine here, but I want to make sure that the user deliberately enters an empty string.  On the command line that looks like "console.php contact add nick http://example.com ''".  If no argument is supplied getArgument returns null and we enter the conditional, but if the argument is an empty string it uses that.

Reply:
`emtpy()` serves the main purpose of checking whether the variable or array key exists before checking for falsy values (`null`/`false`/the number zero/the string zero/empty string). Since you declared the variable the line before, you can skip `empty()` and directly test on what you want. The current test leaves room for 4 falsy values: `null`, `false`, the number zero and the string zero.

It seems that you are testing if the 4th parameter (named `network` in the command synopsis) has not been provided at all, and we can look up the `Console->getArgument()` function for this. It returns `null` by default if the argument wasn't provided, so the condition just becomes
```suggestion
		if ($network === null) {
```